components/sup-protocol/build.rs

```diff
@@ -77,13 +59,7 @@ fn protocol_includes() -> Vec<String> { vec!["protocols".to_string()] }
 fn compile_proto_impls<P>(protos: &[P], includes: &[P]) -> Result<()>
     where P: AsRef<Path>
 {
-    let target: PathBuf = env::var_os("OUT_DIR").ok_or_else(|| {
-                                                    Error::new(ErrorKind::Other,
-                                                               "OUT_DIR environment variable is \
-                                                                not set")
-                                                })?
-                                                .into();
-
+    let target = PathBuf::from("src/generated");
```

Comment:
Why did this protobuf stuff need to get changed in this PR?

(Also, I swear that I've changed this in the past...)

Reply:
@christophermaier see discussion above with @baumanj . Basically it started with the fact that we can't use `rename` to move a file to a mounted directory on windows followed by the question, why don't we just generate the files in place instead of doing a `copy` and `remove_file`

Reply:
Ah, the switch to Docker cause this... gotcha.

We've fixed this before in other crates, but apparently haven't over here 😦 Also, this entire code generation logic is _waaaaaaaaaaaaaaaaay_ more complex than it should be. I'm going to file an issue to fix it up once and for all.

Reply:
Filed #6628